### DISCUSSION TITLE
Nuxt project with prisma deploy to vercel fail

### DISCUSSION THREAD

**Discussion by meron127** [prisma/prisma#17717] [Q&A] [answered]
### Summary

I am new to nuxt and learn by developing a new project which is similar to this [prisma guide](https://www.prisma.io/docs/guides/deployment/deployment-guides/deploying-to-vercel).
The project works fine in local development, but in vercel deployment, I encounter the error.
The error show in logs:
![nuxt-prisma-vercel-example_error](https://user-images.githubusercontent.com/85394063/216226695-c83d9b01-b68b-45b5-b5b4-bfd266a35502.png)

### Additional information

I create the reproduction from [vercel nuxt boilerplate](https://vercel.com/templates/nuxt/nuxtjs-boilerplate) and add some prisma items, add 'DATABASE_URL' environment variable in vercel.
I am using PostgreSQL database by Supabase.

### Example

https://github.com/meron127/nuxt-prisma-vercel-example

**Comment by nurul3101** [comment]
Hey @meron127 👋 

Which command are you using to build your project for deploying to vercel?
Based on your package.json file, are you using `vercel-build` command or just `build` command?

**Reply by meron127** [reply]
`vercel-build` should be used in this deployment.
In the Deployment Status from Vercel, building is successful with prisma migration and nitro server.

**Reply by nurul3101** [reply]
So using `vercel-build` command fixed the issue for you, or are you still running into it?

**Reply by meron127** [reply]
Sorry for the confusion. `vercel build` is already used in the deployment.
Building stage is successful but still error is shown in the production site.

**Reply by nurul3101** [reply]
Did you try clearing the build cache in vercel before deploying?

**Reply by meron127** [reply]
I clear the cache by redeploying. Error is still shown in the logs.

**Reply by nurul3101** [reply]
Can you try just directly deploying the nuxt boilerplate and see if that works as expected?

**Reply by meron127** [reply]
Using the nuxt boilerplate only, it is successful with expected welcome page.

**Comment by meron127** [comment, marked as answer]
Probably find out the solution when reading the [nitro doc](https://nitro.unjs.io/deploy/providers/vercel) accidentally.
The default setting of deployment preset is 'vercel-edge' in the nuxt template provided in vercel.
However, my codes are not using prisma with data proxy (not edge function approach).
After changing deployment preset to 'vercel', it is able to see the homepage and call the api to fetch data from Supabase with prisma.

**Reply by nurul3101** [reply]
Thanks for sharing the solution! 🙌

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
